.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,157 @@
+        run: |
+          sudo apt-get update && sudo apt-get install -y gconf-service libgbm-dev libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libnss3 lsb-release xdg-utils wget ca-certificates
+          sudo apt-get install -y libudev-dev cargo npm imagemagick libmagickwand-dev cmake jq
+      - name: install node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.4.0
+      - name: Install ruby and other related tools
+        run: |
+          whoami
+          npm install -g corepack
+          corepack enable
+          curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+          echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+          apt-get update
+          sudo apt-get install -y yarn
+          yarn --version
+          apt update
+          apt-get install -y software-properties-common
+          apt-add-repository -y ppa:rael-gc/rvm
+          apt-get update
+          apt-get install -y rvm
+          echo 'source "/etc/profile.d/rvm.sh"' >> ~/.bashrc
+          source /etc/profile.d/rvm.sh
+          which rvm
+          rvm install ruby-3.1.2
+          ruby --version
+          sudo chown -R $(whoami) /usr/local
+          rvm use 3.1.2 --default
+          rvm --version
+          rvm info
+          which rvm
+          ruby --version
+      - name: cargo install nj-cli
+        run: |
+          cargo install nj-cli --locked
+          cargo install wasm-pack --locked
+      - name: Prepare environment
+        run: |
+          npm i -g tslib
+          export PATH="/usr/share/rvm:$PATH"
+          which ruby
+          ruby --version
+          pwd
+          gem install dotenv json octokit tmpdir fileutils
+      - name: Get PR details
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/github-script@v7
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          script: |
+            const fs = require('fs');
+            const pr = await github.rest.pulls.get({
+              owner: 'esrlabs',
+              repo: 'chipmunk',
+              pull_number: ${{ github.event.inputs.pr_id }}
+            });
+            const PR_OWNER = pr.data.user.login;
+            const PR_REPO = pr.data.base.repo.name;
+            console.log(`owner is ${PR_OWNER}`);
+            console.log(`repo is ${PR_REPO}`);
+            // Setting environment variables
+            fs.appendFileSync(process.env.GITHUB_ENV, `PR_OWNER=${PR_OWNER}\n`);
+            fs.appendFileSync(process.env.GITHUB_ENV, `PR_REPO=${PR_REPO}\n`);
```

Comment:
is it possible to move script into file? Workflow in general very complicated, would be nice to have a way to make it more simple at least with extracting some scripts (bash, js etc) into files